{{cpp/title|regex_error}}
{{cpp/regex/regex_error/navbar}}
{{ddcl|header=regex|since=c++11|1=
class regex_error;
}}

定义抛出的异常对象类型，以报告正则表达式库中的错误。

{{inheritance_diagram/std-regex_error}}

===成员函数===
{{dsc begin}}
{{dsc inc|cpp/regex/regex_error/dsc regex_error}}
{{dsc inc|cpp/regex/regex_error/dsc operator{{=}}}}
{{dsc inc|cpp/regex/regex_error/dsc code}}
{{dsc end}}

{{cpp/error/runtime_error/inherit}}
{{cpp/error/exception/inherit}}

===示例===
{{example
|code=
#include &lt;iostream&gt;
#include &lt;regex&gt;

int main()
{
    try
    {
        std::regex re("[a-b][a");
    }
    catch (const std::regex_error&amp; e)
    {
        std::cout &lt;&lt; "捕获 regex_error: " &lt;&lt; e.what() &lt;&lt; '\n';
        if (e.code() == std::regex_constants::error_brack)
            std::cout &lt;&lt; "错误码为 error_brack\n";
    }
}
|output=
捕获 regex_error: The expression contained mismatched [ and ].
错误码为 error_brack
|p=true
}}

{{langlinks|de|en|es|fr|it|ja|pt|ru}}